safe-logging/src/main/java/com/palantir/logsafe/Arg.java

```diff
@@ -52,12 +59,13 @@ public final boolean equals(Object other) {
         if (this == other) {
             return true;
         }
-        if (other == null || getClass() != other.getClass()) {
+        if (!(other instanceof Arg)) {
             return false;
         }
         Arg<?> arg = (Arg<?>) other;
         return Objects.equals(name, arg.name)
-                && Objects.equals(value, arg.value);
+                && Objects.equals(value, arg.value)
+                && (isSafeForLogging() == arg.isSafeForLogging());
```

Comment:
this shouldn't change any behaviour because we don't expect people to make their own subclasses of `Arg`, but better safe than sorry!